Problem2.java

```diff
@@ -0,0 +1,53 @@
+/* Problem 2: Search inside rotated sorted array
+*/
+// Time Complexity : O(log n)
+// Space Complexity : O(1) since we are not using any auxiliary space
+// Did this code successfully run on Leetcode : Yes
+// Any problem you faced while coding this : No
+
+
+// Your code here along with comments explaining your approach
+//Follow up question(interviewer): Can we do this in single pass?
```

Comment:
We are doing in single pass only, We will not touch every element but atmost we will touch only once. So it is single pass algorithm only

Reply:
Okay thanks for your feedback.